zenoh/src/api/session.rs

```diff
@@ -449,24 +449,40 @@ impl fmt::Debug for SessionRef<'_> {
     }
 }
 
-/// A trait implemented by types that can be undeclared.
-pub trait Undeclarable<S, O, T = ZResult<()>>
+pub(crate) trait UndeclarableInner<S, O, T = ZResult<()>>
```

Comment:
I would call it `UndeclarableSealed` which is make it clear that follows the *sealed trait* pattern.